### ISSUE TITLE
Make sure tests are passing to complete Docker build

### ISSUE THREAD

**Issue by huard** [bird-house/cookiecutter-birdhouse#84]
Is this a good idea ? This would avoid pushing docker images that fail testing.

**Comment by huard** [comment]
@tlvu @cehbrecht What are your thoughts about this ? Is it overkill ? Necessary ?

**Comment by tlvu** [comment]
We have to test what we ship and docker is our shipping format so yes I think this is necessary.

**Comment by cehbrecht** [comment]
> Is this a good idea ? This would avoid pushing docker images that fail testing.

Can we make the "docker build" an option in the cookiecutter template?

**Comment by tlvu** [comment]
> Can we make the "docker build" an option in the cookiecutter template?

I can look into this once I am done with bringing all the birds up-to-date with the cookiecutter and rolling out cruft as the cookiecutter update mechanism.

**Comment by huard** [comment]
@tlvu Status ?

**Comment by tlvu** [comment]
@huard did not have time to work on this yet.  There is Travis-CI and Monitoring on my mind right now.  How are those compared to this one in terms of priority?

**Comment by huard** [comment]
Depends. I suspect making the change to the cookie-cutter is the easy part. Rolling it to all the birds and fixing broken tests is going to be the hard part, no? Does it make sense to split the job in two ? Can the easy part be done rapidly, and you give me a time estimate for the hard part and I schedule this later in 2021?

**Comment by tlvu** [comment]
> Depends. I suspect making the change to the cookie-cutter is the easy part. 

Some what true.  But I'd rather test on at least one bird before leaving it in the cookie-cutter.  I've also never used this DockerHub feature so I expect some back and forth before getting this dialed.

> Rolling it to all the birds and fixing broken tests is going to be the hard part, no? 

Exact, rolling out, and backport back to cookie-cutter improvements are time consuming.

> Does it make sense to split the job in two ? Can the easy part be done rapidly, and you give me a time estimate for the hard part and I schedule this later in 2021?

Could split the job in two but that feels like leaving the work haft done.  Any birds that will refresh from the cookiecutter will get the change and it might not work for those birds.

We can split the job in multiple incremental releases but all those multiple releases preferably should not be too spaced apart.

If DockerHub support is straight forward, not clashing with how we run tests, maybe 1-2 weeks for cookiecutter and all the birds.  If things get complex, maybe 3-4 weeks.

**Comment by huard** [comment]
Ok, then let's postpone the whole thing. I underestimated the effort and there are other priorities at the moment. Back to the icebox.

**Comment by tlvu** [comment]
Yeah I was giving a conservative estimate here.  I'd rather that than saying something I can not deliver.  This is a bit similar to converting Travis-CI config to another service.  They are both doing similar work but the switch will take some time to be fine tuned.
